test/framework/src/main/java/org/elasticsearch/transport/MockTcpTransport.java

```diff
@@ -370,9 +373,20 @@ public void close() throws IOException {
                     IOUtils.close(serverSocket, activeChannel, () -> IOUtils.close(workerChannels.keySet()),
                         () -> cancellableThreads.cancel("channel closed"), onClose);
                 }
-                assert removedChannel : "Channel was not removed or removed twice?";
+                // assert remoteChannel; is not enough it will throw NPE if removeChannel is null
+                assert removedChannel == Boolean.TRUE: "Channel was not removed or removed twice?";
```

Comment:
I think that `removedChannel != null && removalChannel` is more defensive (here we are relying on either always putting `Boolean.TRUE` in the map (which we do), or the fact that a boxed boolean always boxes to `Boolean.TRUE` (the JVM guarantees this) but I prefer to not rely on these fragile/subtle things).